[Crash] app crashes on  "Get Hired" fragment

**Describe the bug**
App crashes on Get hired fragment

**To Reproduce**

1. Go to navigation drawer
2. Click on 'Get Hired'
3. App crashes

**Logs**
```
2019-11-09 15:29:40.688 3775-3971/com.codingblocks.cbonlineapp.debug E/SQLiteLog: (1) no such table: JobsModel
2019-11-09 15:29:40.701 3775-3971/com.codingblocks.cbonlineapp.debug E/AndroidRuntime: FATAL EXCEPTION: arch_disk_io_1
    Process: com.codingblocks.cbonlineapp.debug, PID: 3775
    java.lang.RuntimeException: Exception while computing database live data.
        at androidx.room.RoomTrackingLiveData$1.run(RoomTrackingLiveData.java:92)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: android.database.sqlite.SQLiteException: no such table: JobsModel (code 1 SQLITE_ERROR): , while compiling: SElECT * FROM JobsModel
        at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
        at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:903)
        at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:514)
        at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:588)
        at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:58)
        at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
        at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:46)
        at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1408)
        at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1383)
        at androidx.sqlite.db.framework.FrameworkSQLiteDatabase.query(FrameworkSQLiteDatabase.java:161)
        at androidx.room.RoomDatabase.query(RoomDatabase.java:328)
        at androidx.room.util.DBUtil.query(DBUtil.java:83)
        at com.codingblocks.cbonlineapp.database.JobsDao_Impl$8.call(JobsDao_Impl.java:582)
        at com.codingblocks.cbonlineapp.database.JobsDao_Impl$8.call(JobsDao_Impl.java:579)

```

**Smartphone Info:**
<!-- Please complete the following information -->
|               | |
|---------------|-|
|Device         |Nokia 6.1|
|Android Version| Android Pie   |

**Would you like to work on the issue?**
<!-- Please let us know if you can work on it or the issue should be assigned to someone else. -->
- [x] Yes
- [ ] No

--- a/app/src/main/java/com/codingblocks/cbonlineapp/jobs/JobsActivity.kt
+++ b/app/src/main/java/com/codingblocks/cbonlineapp/jobs/JobsActivity.kt
@@ -7,8 +7,11 @@ import android.view.View
 import androidx.appcompat.app.AppCompatActivity
 import androidx.appcompat.widget.AppCompatButton
 import androidx.core.view.children
+import androidx.core.view.isVisible
+import androidx.lifecycle.observe
 import androidx.recyclerview.widget.LinearLayoutManager
 import com.codingblocks.cbonlineapp.R
+import com.codingblocks.cbonlineapp.database.models.JobsModel
 import com.codingblocks.cbonlineapp.util.extensions.observer
 import com.codingblocks.cbonlineapp.widgets.SheetDialog
 import com.google.android.material.bottomsheet.BottomSheetBehavior
@@ -18,6 +21,7 @@ import com.google.android.material.chip.ChipGroup
 import com.google.android.material.tabs.TabLayout
 import io.github.inflationx.viewpump.ViewPumpContextWrapper
 import kotlinx.android.synthetic.main.activity_jobs.*
+import org.jetbrains.anko.design.indefiniteSnackbar
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class JobsActivity : AppCompatActivity() {
@@ -44,7 +48,22 @@ class JobsActivity : AppCompatActivity() {
         viewModel.getAllJobs().observer(this) { jobs ->
             viewModel.allJobList.clear()
             viewModel.allJobList.addAll(jobs)
-            jobsAdapter.submitList(viewModel.allJobList)
+
+            /*
+            * To solve the problem with listAdapter.submitList() with referencing
+            * the list. i.e.
+            *
+            *  if(oldList == newList){
+            *       return;
+            *  }
+            *
+            *  therefore, make different object
+             */
+            val updatedJobs = mutableListOf<JobsModel>().apply {
+                addAll(viewModel.allJobList)
+            }
+
+            jobsAdapter.submitList(updatedJobs)
             locationList = jobs.map { it.location }.distinct()
             jobtypeList = jobs.map { it.type }.distinct()
             setupBottomFilterSheet()
@@ -53,6 +72,25 @@ class JobsActivity : AppCompatActivity() {
         viewModel.filteredJobsProgress.observer(this) {
             if (it) {
                 jobsAdapter.notifyDataSetChanged()
+            }
+        }
+
+        viewModel.jobProgress.observe(this) {
+            if (it) {
+                shimmerJobs.isVisible = true
+                rvJobs.isVisible = false
+            } else {
+                shimmerJobs.isVisible = false
+                rvJobs.isVisible = true
+            }
+        }
+
+        viewModel.fetchError.observe(this) {
+            if (it) {
+                shimmerJobs.isVisible = false
+                rootJobs.indefiniteSnackbar("Error Fetching Jobs", "Retry") {
+                    viewModel.getJobs()
+                }
             }
         }
 

--- a/app/src/main/java/com/codingblocks/cbonlineapp/jobs/JobsViewModel.kt
+++ b/app/src/main/java/com/codingblocks/cbonlineapp/jobs/JobsViewModel.kt
@@ -20,14 +20,19 @@ class JobsViewModel(
     var searchFilters = FilterData()
     val allJobList = mutableListOf<JobsModel>()
 
+    var jobProgress = MutableLiveData<Boolean>(false)
+    var fetchError = MutableLiveData<Boolean>(false)
+
     fun getAllJobs() = jobsDao.getAllJobs()
 
     fun getJobs() {
+        jobProgress.value = true
         Clients.onlineV2JsonApi.getJobs(
             getDate(),
             getDate()
-        ).enqueue(retrofitCallback { _, response ->
+        ).enqueue(retrofitCallback { t, response ->
             response?.body().let {
+                fetchError.value = false
                 if (response?.isSuccessful == true) {
                     response.body()?.run {
                         forEach { job ->
@@ -64,8 +69,13 @@ class JobsViewModel(
                                     }
                                 })
                         }
+                        jobProgress.value = false
                     }
                 }
+            }
+
+            t?.let {
+                fetchError.value = true
             }
         })
     }

--- a/app/src/main/res/layout/activity_jobs.xml
+++ b/app/src/main/res/layout/activity_jobs.xml
@@ -4,6 +4,7 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:id="@+id/rootJobs"
     tools:context="com.codingblocks.cbonlineapp.jobs.JobsActivity">
 
     <com.google.android.material.appbar.AppBarLayout
@@ -101,12 +102,31 @@
                 android:src="@drawable/ic_job_filter" />
         </RelativeLayout>
 
-        <androidx.recyclerview.widget.RecyclerView
-            android:id="@+id/rvJobs"
+        <FrameLayout
             android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_margin="8dp"
-            tools:listitem="@layout/item_job" />
+            android:layout_height="wrap_content">
+
+            <com.facebook.shimmer.ShimmerFrameLayout
+                android:id="@+id/shimmerJobs"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content">
+                <LinearLayout
+                    android:orientation="vertical"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content">
+                    <include layout="@layout/item_skeleton_course_card"/>
+                    <include layout="@layout/item_skeleton_course_card"/>
+                </LinearLayout>
+            </com.facebook.shimmer.ShimmerFrameLayout>
+
+            <androidx.recyclerview.widget.RecyclerView
+                android:visibility="gone"
+                android:id="@+id/rvJobs"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_margin="8dp"
+                tools:listitem="@layout/item_job" />
+        </FrameLayout>
 
 
     </LinearLayout>

